### DISCUSSION TITLE
Netbox Bulk Device Upload

### DISCUSSION THREAD

**Discussion by smallboy69** [netbox-community/netbox#15721] [Q&A]
hi Community

am currently trying to bulk upload routers and some switches onto netbox using the bulk upload page where csv and , or autodetect is used...

when i upload the data i get no error message and i cant seem to find the uploaded data...under devices

appreciate if you could please advise where can i look for failed upload with detailed remedial action...as i am new to netbox...

**Comment by a084ed22** [comment]
Could you try to upload a few lines at a time without relying on file upload, and check if that works?

**Reply by smallboy69** [reply]
yes same issue facing....i am using the export template but arranging it as the import variable names as variable names for export and import have different column positions

and when i upload it has  no error message but when i try and check my upload i cant search the dataset i uploaded

just curious can please point me to documentation on variable type and the conditions placed on the variables when importing data


name
role 
tenant 
manufacturer
device_type 
platform
serial
asset_tag
status
site
location
rack
position
face
latitude
longitude
parent
device_bay
airflow
virtual_chassis
vc_position
vc_priority
cluster
description
config_template
comments
tags
id

**Reply by candlerb** [reply]
> yes same issue facing....i am using the export template but arranging it as the import variable names as variable names for export and import have different column positions

No, the column positions are unimportant. But the column *names* are different (especially capitalization, and they are case sensitive). As are the formats of many of the columns.

The CSV export and import are completely different and *not* compatible with each other. You can't use one as a guide to the other.

> and when i upload it has no error message

It would be very helpful if you could give us a sample of the data you're uploading: just the column headers plus one data row.

And also say what version of Netbox you're using, because the behaviour does change over time - e.g. see #13200 / #11617.

> just curious can please point me to documentation on variable type and the conditions placed on the variables when importing data

The documentation is only provided within the upload form. It says what the column headings available are, which ones are mandatory, and a brief description of each.
